lib/matplotlib/axis.py

```diff
@@ -1978,16 +1983,21 @@ def set_ticklabels(self, labels, *, minor=False, fontdict=None, **kwargs):
             func = functools.partial(self._format_with_dict, tickd)
             formatter = mticker.FuncFormatter(func)
         else:
+            _api.warn_external(
+                 "set_ticklabels should only be used with set_ticks or "
+                 "a FixedLocator.")
             formatter = mticker.FixedFormatter(labels)
 
-        if minor:
-            self.set_minor_formatter(formatter)
-            locs = self.get_minorticklocs()
-            ticks = self.get_minor_ticks(len(locs))
-        else:
-            self.set_major_formatter(formatter)
-            locs = self.get_majorticklocs()
-            ticks = self.get_major_ticks(len(locs))
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", message="FixedFormatter should only ")
```

Comment:
Let's be more explicit:
```suggestion
            warnings.filterwarnings(
                "ignore",
                message="FixedFormatter should only be used together with FixedLocator")
```
Also, I'm tempted to move this in to be only around `set_..._formatter`

Reply:
Repeat the filter twice?  Doesn't seem necessary, though it does make it more obvious what call is getting the filter.